index.js

```diff
@@ -332,17 +329,9 @@ async function parsePipelineTemplate({ yaml }) {
     // flatten pipeline template shared setting into jobs
     pipelineTemplateConfig.jobs = flattenSharedIntoJobs(pipelineTemplateConfig.shared, pipelineTemplateConfig.jobs);
 
-    const mergedJobs = {};
+    const { flattenedDoc } = await phaseFlatten(pipelineTemplateConfig, templateFactory);
 
-    // Handle mergeSharedSteps annotation
-    Object.keys(pipelineTemplateConfig.jobs).forEach(j => {
-        mergedJobs[j] = handleMergeSharedStepsAnnotation({
-            sharedConfig: pipelineTemplateConfig.shared,
-            jobConfig: pipelineTemplateConfig.jobs[j]
-        });
-    });
-    delete pipelineTemplateConfig.shared;
-    pipelineTemplateConfig.jobs = mergedJobs;
```

Comment:
so we no longer need to use `handleMergeSharedStepsAnnotation`? if so can you remove this function from the related file? and what is the reason of removing `delete pipelineTemplateConfig.shared;`?